src/modals/WalletConnectModal.tsx

```diff
@@ -0,0 +1,194 @@
+    onClose()
+  }, [walletConnect, proposal, onClose])
+
+  switch (state) {
+    case State.Error:
+      return (
+        <CenteredModal
+          title={<WalletConnectTitle src={walletConnectFull} />}
+          subtitle="Initiate a session with a dApp"
+          onClose={onClose}
+        >
+          <div>Please try to generate a new session identifier in the dApp.</div>
+          <ModalFooterButtons>
+            <ModalFooterButton onClick={onClose}>Close</ModalFooterButton>
+          </ModalFooterButtons>
+        </CenteredModal>
+      )
+    case State.InitiateSession:
+      return (
+        <CenteredModal
+          title={<WalletConnectTitle src={walletConnectFull} />}
+          subtitle="Initiate a session with a dApp"
+          onClose={onClose}
+        >
+          <Input
+            onChange={(t) => setUri(t.target.value)}
+            value={uri}
+            placeholder="Paste what was copied from the dApp"
+          />
+          <ModalFooterButtons>
+            <ModalFooterButton secondary onClick={onClose}>
+              Cancel
+            </ModalFooterButton>
+            <ModalFooterButton onClick={onInitiate} disabled={uri === ''}>
+              Connect
+            </ModalFooterButton>
+          </ModalFooterButtons>
+        </CenteredModal>
+      )
+    case State.Proposal: {
+      const name = proposal?.proposer.metadata.name ?? 'No application name'
+      const url = proposal?.proposer.metadata.url ?? 'No URL specified'
+      const description = proposal?.proposer.metadata.description ?? 'No description given'
+
+      return (
+        <CenteredModal
+          title={<WalletConnectTitle src={walletConnectFull} />}
+          subtitle="Approve the proposal to connect"
+          onClose={onClose}
+        >
+          <Notice>Please review the following before authorizing the dApp</Notice>
+          <List>
+            <Name>{name}</Name>
+            <Url>{url}</Url>
+            <Desc>{description}</Desc>
+          </List>
+          <ModalFooterButtons>
+            <ModalFooterButton secondary onClick={onReject}>
+              Reject
+            </ModalFooterButton>
+            <ModalFooterButton onClick={onApprove}>Approve</ModalFooterButton>
+          </ModalFooterButtons>
+        </CenteredModal>
+      )
+    }
+    case State.RequireUnlock:
+      return (
+        <CenteredModal
+          title={<WalletConnectTitle src={walletConnectFull} />}
+          subtitle="Initiate a session with a dApp"
+          onClose={onClose}
+        >
+          <div>Please unlock a wallet.</div>
+          <ModalFooterButtons>
+            <ModalFooterButton onClick={onClose}>Ok</ModalFooterButton>
+          </ModalFooterButtons>
+        </CenteredModal>
+      )
+    default:
+      return <div>Unknown state</div>
+  }
+}
+
+export default WalletConnectModal
+
+const WalletConnectTitle = styled.img`
+  width: 12rem;
+`
+
+const Notice = styled.div`
+  border: solid 1px #ccc;
```

Comment:
don't we have a theme var for this? (and the following color code)

Reply:
Yeah, the styling for the whole modal is undetermined; it's mostly all placeholder until Mika can get time on it